cluster-autoscaler/processors/provreq/provsioning_request_processor.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provreq
+
+import (
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/provreqclient"
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/provreqwrapper"
+	"k8s.io/client-go/rest"
+)
+
+// ProvisioningRequestProcessor process ProvisionignRequests in the cluster.
+type ProvisioningRequestProcessor interface {
+	Process() error
+	CleanUp()
+}
+
+// ProvisioningClassProcessor process ProvisionignRequests in the cluster.
```

Comment:
I wrote a comment below with full context, will copy the part of it that related to your question: 
"Currently I'm adding support for check-capacity ProvisioningClass, but then I'll add support for atomicScaleUp ProvisioningClass. Each class will process ProvisioningRequest differently, so that's why ProvisioningRequestProcessor has a list of ProvisioningClassProcessor where each processor receives a list of ProvisioningRequests.
Initially I thought to have one interface for ProvisioningRequestProcessor and ProvisioningClassProcessor and wanted to pass nil list to ProvisioningRequestProcessor in the RunOnce() function, but @kisieland asked to have an empty input instead, so I introduced two processors."